### PR TITLE
[V9] Remove regex on FileNamer

### LINKED ISSUE #2586: preg_replace(): Compilation failed: unmatched closing parenthesis at offset 50
This seems to be caused by #2575, this causes issues on filenames like this:
```
[2021-05] Filename.mp4
(2021-05) Filename.mp4
{2021-05} Filename.mp4
```
I don't know if these names are not valid on Windows, but on Linux almost every character is allowed.

```
ErrorException 

  preg_replace(): Compilation failed: unmatched closing parenthesis at offset 50

  at vendor/spatie/laravel-medialibrary/src/Support/FileNamer/FileNamer.php:14
     10▕     public function originalFileName(string $fileName): string
     11▕     {
     12▕         $name = pathinfo($fileName, PATHINFO_FILENAME);
     13▕ 
  ➜  14▕         return preg_replace("/{$name}\..*/", $name, $fileName);
     15▕     }
     16▕ 
     17▕     abstract public function conversionFileName(string $fileName, Conversion $conversion): string;
     18▕ 
```

### PR DESCRIPTION
**Pull request by erikn69**
Closes #2586
@francoism90 please review, suggestions?

### CODE REVIEW

**Comment by francoism90** [comment]
@erikn69 I'm not an expert, why has the change been made? Doesn't Laravel/Symfony offer any solution?

I would simple use a sanitize helper and/or replace all the illegal symbols with `-`.

File: src/Support/FileNamer/FileNamer.php [hunk 1]
```diff
@@ -9,9 +9,9 @@ abstract class FileNamer
 {
     public function originalFileName(string $fileName): string
     {
-        $name = pathinfo($fileName, PATHINFO_FILENAME);
+        $ext_length = strlen(pathinfo($fileName, PATHINFO_EXTENSION));
```
Comment (freekmurze) [inline src/Support/FileNamer/FileNamer.php:12]: Variables are camel cased in all our packages.

File: src/Support/FileNamer/FileNamer.php [hunk 2]
```diff
@@ -9,9 +9,9 @@ abstract class FileNamer
...
 
-        return preg_replace("/{$name}\..*/", $name, $fileName);
+        return substr($fileName, 0, strlen($fileName) - ($ext_length ? $ext_length + 1 : 0));
```
Comment (freekmurze) [inline src/Support/FileNamer/FileNamer.php:14]: To me `strlen($fileName) - ($ext_length ? $ext_length + 1 : 0` is not very readable. Maybe put it the result a new variable with a meaningful name?

File: tests/Feature/FileAdder/IntegrationTest.php [hunk 3]
```diff
@@ -460,14 +460,14 @@ public function the_file_name_can_be_modified_using_custom_sanitizing_and_defaul
     {
         $media = $this->testModel
             ->addMedia($this->getTestJpg())
-            ->usingFileName('other/otherFileName.jpg')
+            ->usingFileName('other/[0]-{0}-(0)-otherFile.Name.jpg')
```
Comment (freekmurze) [inline tests/Feature/FileAdder/IntegrationTest.php:463]: Please do not change the old test, but add a new one.

**Review by freekmurze — CHANGES_REQUESTED** [review]

**Comment by erikn69** [comment]
@freekmurze fixed

**Comment by freekmurze** [comment]
Thanks!

**Pull request merged**